solutions/cf-920E.mdx

```diff
@@ -0,0 +1,450 @@
+### Proof
+
+We can handle all nodes with $\texttt{degree}+1>\dfrac{N}2$ in $O(N)$ time. 
+
+Let there be $N'$ nodes that have at most $\dfrac{N}2 -1 ≈ \dfrac{N}2$ edges. Then, they also must have at least $\dfrac{N}2$ non-edges. Therefore, there must be at least $\dfrac{N'\cdot\dfrac{N}2}2 = \dfrac{N\cdot N'}4$ non-edges (each edge accounts for two nodes).
+
+The input states that the number of edges is less than $2 \cdot 10^5$, therefore $\dfrac{N \cdot N'}4\le2\cdot10^5$, therefore $N \cdot N' \le8\cdot10^5$. 
+
+Doing a simple DFS and checking whether two nodes intersect would be $\mathcal{O}(NN'+ \log M)$, which would fit under the time limit. The $\log M$ comes from processing input. Therefore, this runs under the time limit.
+
+## Implementation
+
+**Time Complexity:** $\mathcal{O}(NN'+ \log M)$
+
+<LanguageSection>
+
+<CPPSection>
+
+```cpp
+#include <bits/stdc++.h>
+using namespace std;
+using vi = vector<int>;
+#define pb push_back
+
+const int MAX_N = 200000;
+unordered_set<int> adj[MAX_N];
+// group[i] = the group of the ith node
+// group[i] = 0 if the ith node hasn't been assigned a group yet
+int group[MAX_N];
+vi currentcomponent; // the nodes in the current component
+// curgroup = the group of the current component
+// curgroup is always either cur or 1
+int curgroup;
+bool vis[MAX_N];
+int n, m;
+// sz[i] = size of group i
+int sz[MAX_N];
+
+void dfs (int x) {
+	// if x can be reached, it is part of the current component
+	currentcomponent.pb(x);
+	vis[x] = true;
+	for (int i = 0; i < n; i++) {
+		// if there is no, non-edge between x and i, i can be visited
+		if (x != i && adj[x].find(i) == adj[x].end()) {
+			// if we can visit a node from group one every node in
+			// the current component can visit group 1 and we dont
+			// have to create a new component
+			if (group[i] == 1) {
+				curgroup = 1;
+				continue;
+			} else {
+				if (!vis[i]) {
+					dfs(i);
+				}
+			}
+		}
+	}
+}
+
+int main() {
+	cin >> n >> m;
+	for (int i = 0; i < m; i++) {
+		int a, b;
+		cin >> a >> b;
+		a--;
+		b--;
+		adj[a].insert(b);
+		adj[b].insert(a);
+	}
+	int cur = 1; // current number of groups + 1
+	for (int i = 0; i < n; i++) {
+		// if any two nodes have degree+1 > n/2, they are part of the same component
+		if (n - adj[i].size() > n / 2) {
+			group[i] = 1; // assign ith node to component 1
+			sz[1]++; // increase the size of component 1
+			
+			cur = 2; // there is at least one group
+		}
+	}
+	for (int i = 0; i < n; i++) {
+		if (!group[i]) {
+			// curgroup's default is cur
+			curgroup = cur;
+			dfs(i);
+			for (int j : currentcomponent) {
+				group[j] = curgroup;
+				sz[curgroup]++;
+			}
+			currentcomponent.clear();
+			// if a new group has been created, e.g. curgroup = cur, increase cur
+			if (curgroup == cur) {
+				cur++;
+			}
+		}
+	}
+	cout << cur - 1 << endl;
+	sort(sz, sz + cur);
+	for (int i = 1; i < cur; i++) {
+		cout << sz[i] << ' ';
+	}
+	cout << endl;
+}
+
+```
+</CPPSection>
+
+</LanguageSection>
+---
```

Comment:
looks like the file got duplicated and everything below should be deleted?

Reply:
my bad, yes